### DISCUSSION TITLE
Sketch for a future collection role

### DISCUSSION THREAD

**Discussion by Kariton** [evrardjp/ansible-keepalived#222] [Ideas]
~How can we provide useful informations and examples for different use cases or ways to utilize this role?~

~I think about a structure as following:~
~- examples~
   ~- case~

Something like this:
https://github.com/Kariton/ansible-keepalived/tree/play_examples/examples/lvs
(is WIP)

**Comment by evrardjp** [comment]
I think it's a great idea. 
The collections mechanism has the feature of sharing playbooks.
For me, it would make sense that we follow the collection structure for that (and not re-invent the wheel). 
In the meantime, tests contains the structure too... 
WDYT?

**Reply by Kariton** [reply]
For now i dont know anything about galaxy functions or structures and havent looked into those in detail.
But IMO youre right. This should be the target.

I know about collections and roles and that collections are the preferred way of doing things.
But ATM I only consume galaxy content (or directly from git) in a _flat root_ CLI Structure (all my ``playbooks`` in one place - ``roles`` in another and so on). Its a mess.

I know the downsides of my approach and currently try to sort my stuff and started with testing on my own roles.
Until a few days ago i havent looked into linters in general.

The next thing is requirements and separation into ""projects"" which will utilize different collections / roles and provides ``playbooks``.

So im on my way to learn about these practices in general.

I thought about your ``test`` folder and actively decided against it as this is directly related to working tests for molecule as my is more of a writeup.

After all, im looking forward to work with you and support your target where i can.

**Reply by evrardjp** [reply]
> After all, im looking forward to work with you and support your target where i can.

Oh that's so lovely to hear.

Same on my side. 

I feel like collections are the intended place for sharing playbooks.
So it makes sense to me to add this into a collection. 

Let's discuss this in #219 .
